pipelines/batch/src/main/java/com/google/fhir/analytics/FhirSearchUtil.java

```diff
@@ -120,26 +115,7 @@ String findBaseSearchUrl(Bundle searchBundle) {
       throw new IllegalArgumentException(
           String.format("No proper link information in bundle %s", searchBundle));
     }
-
-    try {
-      URI searchUri = new URI(searchLink);
-      NameValuePair pagesParam = null;
-      for (NameValuePair pair : URLEncodedUtils.parse(searchUri, StandardCharsets.UTF_8)) {
-        if (pair.getName().equals("_getpages")) {
```

Comment:
The problem with dropping the `_getpages` parameter is that it breaks OpenMRS and maybe some other EMRs that use the [HAPI FHIR Plain server](https://hapifhir.io/hapi-fhir/docs/server_plain/introduction.html). I just tried a query without the `_getpages` argument and with `_offset` (and `_count`) against an OpenMRS instance and it returns a very large payload (I guess somehow the pagination is disabled). This might be a bug in OpenMRS's implementation but I want us to first figure/fix that. I think that is the root cause for the e2e test failure as well (the payload is large hence it times out).

Here is my suggestion to fix the issue you are facing while not breaking OpenMRS tests: Check the arguments and if `_getpages` is available use the old logic. If `_getpages` is missing use the new `_offset` based approach you have implemented in this PR.

Reply:
That does seem like a fair approach. We will investigate it and thank you for the feedback.

The reason we removed  the `_getpages` and went with the `_offset` is because it was my understanding that the `_getpages` isn't technically part of the FHIR specification.